plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/OptInPopup.java

```diff
@@ -0,0 +1,105 @@
+package com.google.cloud.tools.eclipse.usagetracker;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Font;
+import org.eclipse.swt.graphics.FontData;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Shell;
+
+/**
+ * A one-time dialog to suggest opt-in for sending client-side usage metrics.
+ */
+class OptInDialog extends Dialog {
+
+  public OptInDialog(Shell parentShell) {
+    super(parentShell);
+    setShellStyle(SWT.MODELESS);
+    setBlockOnOpen(false);
+  }
+
+  /**
+   * Show this dialog at the top-right corner.
+   */
+  @Override
+  protected Point getInitialLocation(Point initialSize) {
+    Rectangle parentBounds = getParentShell().getBounds();
+    Rectangle parentClientArea = getParentShell().getClientArea();
+
+    int heightCaptionAndUpperBorder =
+        parentBounds.height - parentClientArea.height - getParentShell().getBorderWidth();
+    return new Point(parentBounds.x + parentClientArea.width - initialSize.x,
+        parentBounds.y + heightCaptionAndUpperBorder);
+  }
+
+  /**
+   * Overridden in order to remove the button bar area (and the OK and CANCEL buttons) that the
```

Comment:
Let's talk offline about the UI. I think we can add the buttons and make this a lot clearer to the user. We don't want anyone opting in by accident.